src/shapes/Object/StackedObject.ts

```diff
@@ -49,7 +48,8 @@ export class StackedObject<
     while (parent) {
       if (target === parent) {
         return true;
-      } else if (parent instanceof fabric.StaticCanvas) {
+        //@TODO ideally find a different way to identify a canvas without importing it
```

Comment:
I don't think this is a big deal.
Canvas is imported in every app so why does it matter?

Reply:
```suggestion

```

Reply:
It does not matter 100%.
But for example i have an use case in which i want to just produce text  pngs and i can do it without a canvas instance.
It would be great if fabric was that flexible.

> IMO env should be a class with an exported instance, same as config for the following reasons:
> 
> * test stubbing/mocking is easy (no need for `setEnvForTests`, an object is better to export in this case than multiple exports)
the reason why i added multiple exports is because fabric.getDocumnet is nicere than fabric.getEnv().document.
I think that getEnv() is a stop gap solution and until we make bundling work we shouldn't think we either need it or we need a different one.

i think the order of priority should be:
bundling > removing ts-nochecks > node vs browser imports > those details here.